src/github3/session.py

```diff
@@ -137,6 +160,10 @@ def oauth2_auth(self, client_id, client_secret):
 
     def request(self, *args, **kwargs):
         """Make a request, count it, and handle 2FA if necessary."""
+        kwargs = kwargs.copy()
```

Comment:
I'm curious why you chose to copy these here.

Since these are collected by the function signature, I wonder what you think about:

```suggestion
kwargs.setdefault('timeout', (self.default_connect_timeout, self.default_read_timeout))
```

Also what do you think of adding a property, e.g.,

```py
@property
def timeout(self):
    return (self.default_connect_timeout, self.default_read_timeout)
```

Reply:
The copy was an overabundance of caution out of habit, but I just checked, and even if you call the method with an existing arg dict and the ** operator (as in: request(**my_kwargs)) you still get a new dictionary.  I learned a new thing about python.  :)

I'll update it to skip the extra copy and add the property.